fair_svm/fair_svm.py

```diff
@@ -0,0 +1,356 @@
+    support_ : array-like, shape = [n_SV]
+        Indices of support vectors.
+
+    support_vectors_ : array-like, shape = [n_SV, n_features]
+        Support vectors.
+
+    dual_coef_ : array, shape = [n_class-1, n_SV]
+        Coefficients of the support vector in the decision function.
+        For multiclass, coefficient for all 1-vs-1 classifiers.
+        The layout of the coefficients in the multiclass case is somewhat
+        non-trivial. See the section about multi-class classification in the
+        SVM section of the User Guide for details.
+
+    coef_ : array, shape = [n_class-1, n_features]
+        Weights assigned to the features (coefficients in the primal
+        problem). This is only available in the case of a linear kernel.
+
+        `coef_` is a readonly property derived from `dual_coef_` and
+        `support_vectors_`.
+
+    intercept_ : array, shape = [n_class * (n_class-1) / 2]
+        Constants in decision function.
+
+
+    """
+
+    def __init__(self, C=1.0, kernel='rbf', degree=3, gamma='auto',
+                 coef0=0.0, 
+                 tol=1e-3,  class_weight=None,
+                 max_iter=1000, 
+                 random_state=None):
+
+        self.C = C
+        self.kernel = kernel
+        self.degree = degree
+        self.gamma = gamma
+        self.coef0 = coef0
+        self.tol = tol
+        self.class_weight = class_weight
+        self.max_iter = max_iter
+        self.random_state = random_state
+
+        np.random.seed(self.random_state)
+
+        self.gram_matrix = None # gram matrix will be computed in the fit method
+        
+
+
+    def compute_and_save_kernel(self, X1, X2):
+
+       # # if the kernel is a callable, then compute it
+        if callable(self.kernel):            
+            gram_matrix = self.kernel(X1, X2)
+            return gram_matrix
+
+
+        # TODO: Check all the kernels again
+        ## Compute the kernel
+        if self.kernel == "linear":
+            gram_matrix = linear_kernel(X1, X2)
+        elif self.kernel == "poly":
+            gram_matrix = polynomial_kernel(X1, X2, degree = self.degree, gamma=self.gamma, coef0 = self.coef0)
+        elif self.kernel == "rbf":
+            gram_matrix = rbf_kernel(X1, X2, gamma = self.gamma)
+        elif self.kernel == "sigmoid":
+            gram_matrix = sigmoid_kernel(X1, X2, gamma = self.gamma, coef0 = self.coef0)
+
+        return gram_matrix
+
+    def fit(self, X, y, x_sensitive, apply_cons = False, cov_thresh=0):
+        """
+        TODO
+        A reference implementation of a fitting function
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape = [n_samples, n_features]
+            The training input samples.
+        y : array-like, shape = [n_samples] or [n_samples, n_outputs]
+            The target values (class labels in classification, real numbers in
+            regression).
+        x_sensitive: the sensitive attribute. has to be categorical
```

Comment:
This approach is different from the one used by `FairLogitEstimator`-- we should discuss pros/cons of passing the sensitive attribute data separately, the way you're doing here, vs. passing it together with the rest of the training data and indicating which columns are sensitive (like `FairLogitEstimator` does).